### DISCUSSION TITLE
Problem with Tracking Foreign Key Deletion with Javers

### DISCUSSION THREAD

**Discussion by ZMikh** [javers/javers#1384] [Q&A]
I am using Javers to audit changes in my entities. However, Javers does not seem to track the deletion of a foreign key reference when I set the reference to null and save the entity. Here are the details.

Entities
I have two entities: **User** and **Profile**. The User entity has a one-to-one foreign key reference to Profile. I have annotated the profile field with `@ValueObject` to indicate that it should be tracked as a value object by Javers.
In my service, I set the User's profile field to null and save the User entity, then commit the changes to Javers.

Issue
When I set the profile field to null and save the User entity, Javers does not seem to register this change. Additionally, changes within the Profile entity itself are tracked correctly, but when the profile reference is set to null, no changes are shown.
When checking the `changed_properties` column in the `jv_snapshot` table, I can see that the `["profile"]` is present. However, when I call **findChanges()** using the query builder, it does not return the expected changes related to setting the profile property to null.
Here is the query:
```
JqlQuery query = QueryBuilder.byInstanceId(user.getId(), User.class)
            .withChildValueObjects()
            .build();
javers.findChanges(query);
```

Environment
 • Javers version: 6.14.0
 • Spring version: 2.7.18
 • Java version: 17
If there's any additional configuration or annotations needed to make Javers track these changes correctly, please let me know.
Thank you for your help!

**Comment by bartoszwalacik** [comment]
See if this helps https://github.com/javers/javers/issues/1358

**Reply by ZMikh** [reply]
Thank you for your prompt response. 

We have reviewed the ticket you provided https://github.com/javers/javers/issues/1358; however, the reported issue is not related to our current problem.
Furthermore, we have upgraded our library version to **7.4.4** as recommended, along with Spring to **3.3.0**, and Java to **17**. Despite these upgrades, the behavior remains unchanged.

We will continue to focus on resolving our issue. Your assistance in helping us would be highly appreciated. 
Thank you for your help.

**Comment by bartoszwalacik** [comment]
@ZMikh check the current situation of https://github.com/javers/javers/issues/1358, I'm currently releasing 7.4.5. besides that, prepare a failing test case (see https://github.com/javers/javers/blob/master/CONTRIBUTING.md)
